src/server/game/Entities/Player/Player.cpp

```diff
@@ -14882,16 +14882,24 @@ void Player::RewardQuest(Quest const* quest, uint32 reward, Object* questGiver,
     uint32 quest_id = quest->GetQuestId();
 
     for (uint8 i = 0; i < QUEST_ITEM_OBJECTIVES_COUNT; ++i)
-        if (quest->RequiredItemId[i])
-            DestroyItemCount(quest->RequiredItemId[i], quest->RequiredItemCount[i], true);
-
+    {
+        if (ItemTemplate const* itemTemplate = sObjectMgr->GetItemTemplate(quest->RequiredItemId[i]))
+        {
+            if (quest->RequiredItemCount[i] > 0 && (itemTemplate->Bonding == BIND_QUEST_ITEM || itemTemplate->Bonding == BIND_QUEST_ITEM1))
```

Comment:
Eff me. I removed the wrong one locally, fixed it, and forgot to commit. Anyway, I will fix it again...

As for the second parameter, there are two DestroyItemCount methods:
https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/game/Entities/Player/Player.cpp#L12346
and
https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/game/Entities/Player/Player.cpp#L12346

The second one only checks inventory (as you can see). I am not sure why both are needed, but that was outside the scope of this PR.